### PR TITLE
fix: load class controller should skip getter & setter

### PR DESCRIPTION
**Pull request by gxcsoccer**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: lib/loader/mixin/controller.js [hunk 1]
```diff
@@ -54,15 +54,18 @@ module.exports = {
...
     // getOwnPropertyNames will return constructor
     // that should be ignored
     if (key === 'constructor') {
       continue;
     }
-    if (is.function(Controller.prototype[key])) {
+    // skip getter, setter & non-function properties
+    const d = Object.getOwnPropertyDescriptor(proto, key);
+    if (!d.get && !d.set && is.function(Controller.prototype[key])) {
       ret[key] = methodToMiddleware(Controller, key);
```
Comment (atian25) [inline lib/loader/mixin/controller.js]: proto[key]

File: lib/loader/mixin/controller.js [hunk 2]
```diff
@@ -54,15 +54,18 @@ module.exports = {
...
   for (const key of keys) {
     // getOwnPropertyNames will return constructor
     // that should be ignored
     if (key === 'constructor') {
       continue;
     }
-    if (is.function(Controller.prototype[key])) {
+    // skip getter, setter & non-function properties
+    const d = Object.getOwnPropertyDescriptor(proto, key);
+    if (!d.get && !d.set && is.function(Controller.prototype[key])) {
```
Comment (popomore) [inline lib/loader/mixin/controller.js]: 直接取 value 吧

Comment (gxcsoccer) [inline lib/loader/mixin/controller.js]: good idea

**Comment by gxcsoccer** [comment]
\+ egg-core@3.9.0

**Pull request merged**